src/commands/transaction/mosaicaddressrestriction.ts

```diff
@@ -0,0 +1,92 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import chalk from 'chalk';
+import { command, metadata, option } from 'clime';
+import {
+    Deadline,
+    MosaicAddressRestrictionTransaction,
+    UInt64,
+} from 'nem2-sdk';
+import { AnnounceTransactionsCommand, AnnounceTransactionsOptions } from '../../announce.transactions.command';
+import { TargetAddressResolver } from '../../resolvers/address.resolver';
+import { MosaicIdAliasResolver } from '../../resolvers/mosaic.resolver';
+import { RestrictionKeyResolver } from '../../resolvers/restrictionKey.resolver';
+import { RestrictionValueResolver } from '../../resolvers/restrictionValue.resolver';
+import { NumericStringValidator } from '../../validators/numericString.validator';
+
+export class CommandOptions extends AnnounceTransactionsOptions {
+    @option({
+        flag: 'i',
+        description: 'Identifier of the mosaic being restricted.',
+    })
+    mosaicId: string;
+
+    @option({
+        flag: 'a',
+        description: 'Address being restricted.',
+    })
+    targetAddress: string;
+
+    @option({
+        flag: 'k',
+        description: 'Restriction key relative to the reference mosaic identifier.',
+    })
+    restrictionKey: string;
+
+    @option({
+        flag: 'V',
+        description: 'New restriction value.',
+        validator: new NumericStringValidator(),
+    })
+    newRestrictionValue: string;
+}
+
+@command({
+    description: 'Set a mosaic restriction to an specific address',
+})
+export default class extends AnnounceTransactionsCommand {
+    constructor() {
+        super();
+    }
+    @metadata
+    execute(options: CommandOptions) {
+        const profile = this.getProfile(options);
+        const account = profile.decrypt(options);
+        const mosaicId = new MosaicIdAliasResolver().resolve(options);
+        const targetAddress = new TargetAddressResolver().resolve(options);
+        const restrictionKey = new RestrictionKeyResolver().resolve(options);
+        const newRestrictionValue = new RestrictionValueResolver().resolve(options);
+
+        const mosaicAddressRestrictionTransaction = MosaicAddressRestrictionTransaction.create(
+            Deadline.create(),
+            mosaicId,
+            restrictionKey,
+            targetAddress,
+            newRestrictionValue,
+            profile.networkType,
+            undefined,
+            UInt64.fromNumericString(options.maxFee),
+        );
+
+        const networkGenerationHash = profile.networkGenerationHash;
+        const signedTransaction = account.sign(mosaicAddressRestrictionTransaction, networkGenerationHash);
+        console.log(chalk.green('signed transaction hash: \n'));
+        console.log(signedTransaction.hash + '\n');
```

Comment:
``announceTransaction`` already prints the signed transaction hash:

https://github.com/nemtech/nem2-cli/blob/072cb578e09488284ecd34e0b72f860ed66dbb37/src/announce.transactions.command.ts#L51-L54